helpers/check_ip_in_list.js

```diff
@@ -26,51 +26,22 @@ var ip = require('ip');
  * @param {boolean} returnListIsEmpty - The return value, if list is empty
  * @returns {boolean} True if ip is in the list, false otherwise
  */
-function CheckIpInList(list, addr, returnListIsEmpty) {
+function CheckIpInList(list, addr) {
 	var i;
-	var n;
-
-	if (!_.isBoolean(returnListIsEmpty)) {
-		returnListIsEmpty = true;
-	}
-
 	if (!_.isArray(list) || list.length === 0) {
-		return returnListIsEmpty;
+		return false;
 	}
-
-	if (!list._subNets) {
-		// First call, create subnet list
-		list._subNets = [];
-		for (i = list.length - 1; i >= 0; i--) {
-			var entry = list[i];
-			if (ip.isV4Format(entry)) {
-				// IPv4 host entry
-				entry += '/32';
-			} else if (ip.isV6Format(entry)) {
-				// IPv6 host entry
-				entry += '/128';
-			}
-			try {
-				var subnet = ip.cidrSubnet(entry);
-				list._subNets.push(subnet);
-			} catch (err) {
-				console.error('CheckIpInList:', err.toString());
-			}
-		}
+	if (!ip.isV4Format(addr)) {
+		throw new TypeError('Provided address is not in IPv4 format');
 	}
-
-	if (list._subNets.length === 0) {
-		return returnListIsEmpty;
-	}
-
-	// Check subnets
-	for (i = 0, n = list._subNets.length; i < n; i++) {
-		if (list._subNets[i].contains(addr)) {
+	for (i = 0; i < list.length; i++) {
+		if (!ip.isV4Format(list[i])) {
+			var message = `Entry with wrong format in list of IPs: ${list[i]}`;
+			console.error('CheckIpInList:', message);
+		} else if (ip.isEqual(list[i], addr)) {
```

Comment:
`ip.isEqual` !== `ip.contains`. Refer to npm ip module and `cidrSubnet` for details.